src/estimation/include/iDynTree/Estimation/AttitudeEstimator.h

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (C) 2018 Fondazione Istituto Italiano di Tecnologia
+ *
+ * Licensed under either the GNU Lesser General Public License v3.0 :
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ * or the GNU Lesser General Public License v2.1 :
+ * https://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ * at your option.
+ */
+
+#ifndef IDYNTREE_ATTITUDEESTIMATOR_H
+#define IDYNTREE_ATTITUDEESTIMATOR_H
+
+#include <iDynTree/Core/VectorFixSize.h>
+#include <iDynTree/Core/Rotation.h>
+
+#include <iostream>
+
+namespace iDynTree
+{
+    typedef iDynTree::Vector3 LinearAccelerometerMeasurements;
+    typedef iDynTree::Vector3 GyroscopeMeasurements;
+    typedef iDynTree::Vector3 MagnetometerMeasurements;
+
+    typedef iDynTree::Vector4 Quaternion;
+    typedef iDynTree::Vector3 RPY;
+
+    /**
+     * @class IAttitudeEstimator generic interface for attitude estimator classes
+     *
+     * The aim is to implement different attitude estimators as a block that takes IMU measurements
+     * as inputs and gives attitude estimates as outputs. This way the underlying implementation is abstracted
+     * and the user only has to set a few parameters and run the estimator.
+     *
+     * The general procedure to use the estimators would be,
+     * - instantiate the filter,
+     * - set initial internal state
+     * - in a loop,
+     *     - update the filter with measurements
+     *     - propagate the states
+     *
+     * However, additional methods to set and get parameters for the filter might be available with respect to the filters.
+     */
+    class IAttitudeEstimator
+    {
+    public:
+        virtual ~IAttitudeEstimator();
+
+        /**
+         * @brief Update the filter with accelerometer and gyroscope measurements
+         *
+         * @param[in] linAccMeas left trivialized 3D vector of linear proper sensor acceleration measuements
```

Comment:
"left trivialized" has a precise meaning, that only make sense for non-commutative lie groups, so it does not make sense to refer to magnetometers measurements or accelerometer measurements that are expressed in body frame as "left trivialized". I guess you could try to uniform the description of the parameters with what you can find https://github.com/robotology/idyntree/blob/master/src/estimation/include/iDynTree/Estimation/ExtWrenchesAndJointTorquesEstimator.h#L176 , but to be honest also that description seems quite obscure to me. Probably it does not make sense to work on this in this PR, but  the most clear option is probably to create a specific page in iDynTree documentation summarizing a few definition that we use in iDynTree, and refer it from there. For example, if you have already introduced ${}^A o_S \in \mathbb{R}^3$ (position vector of the sensor frame w.r.t. an inertial frame and the constant gravity vector ${}^A g \in \mathbb{R}^3$, the quantity you are interested can be documented unambiguously as:  
$$
linAccMeas = {}^S R_A ( {}^A \ddot{o}_A - {}^A g) 
$$
that I think we can agree is better than any possible jargon : )